runtime/util/romclasswalk.c

```diff
@@ -693,14 +709,20 @@ static void allSlotsInConstantPoolDo(J9ROMClass* romClass, J9ROMClassWalkCallbac
 				callbacks->slotCallback(romClass, J9ROM_U32, &((J9ROMMethodHandleRef *)&constantPool[index])->handleTypeAndCpType, "cpFieldHandleTypeAndCpType", userData);
 				break;
 
+			case J9CPTYPE_CONSTANT_DYNAMIC:
+				callbacks->slotCallback(romClass, J9ROM_NAS, &((J9ROMConstantDynamicRef *)&constantPool[index])->nameAndSignature, "cpFieldNAS", userData);
+				callbacks->slotCallback(romClass, J9ROM_U32, &((J9ROMConstantDynamicRef *)&constantPool[index])->bsmIndexAndCpType, "cpFieldBSMIndexAndCpType", userData);
+				break;
+
 			case J9CPTYPE_UNUSED:
 			case J9CPTYPE_UNUSED8:
 				callbacks->slotCallback(romClass, J9ROM_U64, &constantPool[index], "cpFieldUnused", userData);
 				break;
 
 			default:
-				/* unknown cp type - bail */
-				return;
+				/* unknown cp type */
```

Comment:
This is an error path, did you actually detect this path being taken?

Reply:
Not sure; I would have to check that specifically. It definitely would be taken for any class that has constant dynamic CP entries. Now that all the CP entry types are handled properly, this path should never be taken. If in the future new CP entry types are added but not handled by this code, I think it's better to visit them here as "unknown" rather than silently skipping the rest of the constant pool.

Reply:
Okay, my preference would be to add an assertion so we know there is a problem.

Reply:
`Assert_Util_unreachable`

Reply:
Sure. I'll make the changes.